### ISSUE TITLE
native-base breaks with RN 0.58.0

### ISSUE THREAD

**Issue by hoboman313** [GeekyAnts/NativeBase#2534]
I'm seeing an `Invariant violation... is not usable as a native method argument` exception stemming from native-base's Header component.
I believe the breakage is due to: `Public methods of components converted to ES6 classes are no longer bound to their component instance. For ScrollView, the affected methods are setNativeProps, getScrollResponder, getScrollableNode, getInnerViewNode, scrollTo, scrollToEnd, scrollWithoutAnimationTo, and flashScrollIndicators. For CameraRollView, the affected methods are: rendererChanged. For SwipeableRow, the affected methods are: close. Therefore, it is no longer safe to pass these method by reference as callbacks to functions. Auto-binding methods to component instances was a behaviour of createReactClass that we decided to not preserve when switching over to ES6 classes.`
I see native-base's component code use a multitude of these breaking functions.

**Comment by suvenduchhatoi** [comment]
@hoboman313 
See if the discussion in #2536 helps you fix your issue... 
![screen shot 1440-05-24 at 12 37 15 pm](https://user-images.githubusercontent.com/44160581/51964308-1f42be80-248c-11e9-82fe-3f9edc8ff299.png)
![screen shot 1440-05-24 at 12 39 15 pm](https://user-images.githubusercontent.com/44160581/51964309-1f42be80-248c-11e9-9499-83b715425ad6.png)
If not, share your snippet and help us reproduce your issue...

**Comment by ggdegreat** [comment]
2.11.0 doesn't fixed.

**Comment by suvenduchhatoi** [comment]
@ggdegreat Let us know what error you are getting... It's working fine...
![screen shot 1440-05-26 at 12 38 50 pm](https://user-images.githubusercontent.com/44160581/52107984-898b6880-261e-11e9-8805-68ba4d74508f.png)
![screen shot 1440-05-26 at 12 39 01 pm](https://user-images.githubusercontent.com/44160581/52107985-898b6880-261e-11e9-87e0-c97b07053901.png)

**Comment by hoboman313** [comment]
It turned out that my error wasn't related to native-base (I think). I think it might have had something to do with passing null/undefined styles. I got around the issue by tweaking the styles. Your specific error message should tell you what is NaN to give you a hint.

**Comment by wuno** [comment]
#2536 was related to, `react-timer-mixin` which was a problem I was facing. Now that was fixed in 2.11.0 by adding it as a dependency in Native-Base but on a fresh install this new error is shown. 

Specially I am seeing an error on IOS which is -

> height":"<<NaN>> ...  is not usable as a native method argument. 

It goes on to explain this is coming from the native-base Header component. When I check the ejected Header component I can see this exact line it is referencing in the error log for IOS, 

```
height:
      variables.platform === "ios" && variables.platformStyle === "material"
        ? variables.toolbarHeight + StatusBar.height
        : variables.toolbarHeight,
```

This is my component which is using the Header component that is throwing the error -

```
import React from 'react';
import { Body, Button, Header, Icon, Left, Right, Title } from 'native-base';

const AuthHeader = props => {
  return (
    <Header>
      <Left>
        <Button transparent onPress={() => props.navigation.toggleDrawer()}>
          <Icon name="menu" />
        </Button>
      </Left>
      <Body>
        <Title>Header</Title>
      </Body>
      <Right>
        <Button transparent onPress={() => props.navigation.toggleDrawer()}>
          <Icon name="menu" />
        </Button>
      </Right>
    </Header>
  );
};

export default AuthHeader;

```

I am not using a `StatusBar` and confident that is what is causing the issue. 

When I change the Header component from, 

`StatusBar.height` to `1`, this clears the NaN error and the application loads as normal. 

This leaves me with the question,

A. Am I supposed to us use a `StatusBar` with the Native-Base `Header` component? 
B. Or, should there be a check to make sure `StatusBar` does not exists before relying on it's height in the Native-Base `Header` component?

**Comment by ggdegreat** [comment]
@hoboman313 i didn't do anything. it is just a basic call to ActionSheet.

**Comment by suvenduchhatoi** [comment]
@ggdegreat https://github.com/GeekyAnts/NativeBase/issues/2534#issuecomment-459706097 
https://github.com/GeekyAnts/NativeBase/issues/2545 
![screen shot 1440-05-26 at 6 29 49 pm](https://user-images.githubusercontent.com/44160581/52193275-6fe35e80-2874-11e9-9e04-d6ea1d97a718.png)

**Comment by suvenduchhatoi** [comment]
@wuno https://github.com/GeekyAnts/NativeBase/issues/2534#issuecomment-459931809 We are going through this, please make a separate issue for this...

**Comment by aligenc** [comment]
fixed up with 2.11.0

**Comment by janpieterz** [comment]
I'm still seeing this with 2.11.0 🤔

**Comment by wuno** [comment]
This is a different problem than what 2.11.0 fixes. Unless I am missing something here, 2.11.0 adds, 

`"react-timer-mixin": "^0.13.4"` To the package.json. 

The problem mentioned in this thread and in the point I was making has to do with the Header component that occurs when using the material theme on IOS it looks like. Possibly more depending on the original posted setup. 

If you look at my comment above it shows the exact errorI was getting. I was able to fix it for now by adding a default height to status bar when it does not exist. That way it does not throw an error. This is the only issue I am facing now and was able to fix it till a PR is made. 

I would be happy to make a PR but I am not sure what the best long term fix is due to my lack of knowledge of Native Base. This is my first time using it.

**Comment by suvenduchhatoi** [comment]
@wuno Same issue in #2581. Let's discuss it there

**Comment by kala888** [comment]
@wuno same issue, https://github.com/GeekyAnts/NativeBase/issues/2647

**Comment by rajnishcoder** [comment]
I tried 2.11.0 as @SupriyaKalghatgi said but doesn't worked
tried 2.12.0 and 2.12.1 working but height issue, it is showing something like 20px height no matter even if you manually define `height, minHeight, maxHeight`  

BTW, I just love the issue resolving process of git,
@kala888 closed this issue and refer to [2581](https://github.com/GeekyAnts/NativeBase/issues/2581)
Then closed 2581 and ref to [2647](https://github.com/GeekyAnts/NativeBase/issues/2647)
Then closed 2647 and ref to [2720](https://github.com/GeekyAnts/NativeBase/issues/2720)

let see the progress of 2720 and  if this is Recursion or something else  :laughing:

**Comment by silvainSayduck** [comment]
I get `Malformed calls from JS: field sizes are different.` when trying to open an `ActionSheet` too. I'm using `native-base 2.12.1` and `react-native 0.59.8`.

**Comment by davidfarinha** [comment]
I faced this error with react-native 0.59.4 + native-base 2.12.1 when trying to clear padding on a header containing a search bar:
```
<Header searchBar style={[{ ,paddingLeft:0,paddingTop:0,paddingBottom:0,paddingRight:0,backgroundColor: "#53bdfd" }, props.headerStyle]}>
```
Specifically the `paddingTop:0` part, if I removed that it ran without issues.
Managed to fix this error by upgrading to native-base 2.13.4.
